sawmill-core/src/main/java/io/logz/sawmill/processors/UrlDecodeProcessor.java

```diff
@@ -0,0 +1,108 @@
+package io.logz.sawmill.processors;
+
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.ProcessResult;
+import io.logz.sawmill.Processor;
+import io.logz.sawmill.annotations.ProcessorProvider;
+import io.logz.sawmill.exceptions.ProcessorConfigurationException;
+import io.logz.sawmill.exceptions.ProcessorExecutionException;
+import io.logz.sawmill.utilities.JsonUtils;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@ProcessorProvider(type = "urlDecode", factory = UrlDecodeProcessor.Factory.class)
+public class UrlDecodeProcessor implements Processor {
+
+    private Boolean allFields;
+    private String charSet;
+    private String field;
+    private List<String> tagOnFailure;
+
+    public UrlDecodeProcessor(Boolean allFields, String charSet, String field, List<String> tagOnFailure) {
+        this.allFields = allFields;
+        this.charSet = charSet;
+        this.field = field;
+        this.tagOnFailure = tagOnFailure;
+    }
+
+    @Override
+    public ProcessResult process(Doc doc) throws InterruptedException {
+        Map<String,Object> map = doc.getSource();
+        try{
+            if(allFields) {
+                urlDecodeMap(map);
+            }else if(doc.hasField(field)){
+                doc.replaceFieldValue(field,decodeUrl(doc.getField(field)));
+            }
+        }
+        catch(Exception ex){
+            doc.appendList("tags", tagOnFailure);
+            if(ex instanceof InterruptedException) throw new InterruptedException();
+            return ProcessResult.failure("failed to url decode url", new ProcessorExecutionException("urlDecode", ex));
+        }
+        return ProcessResult.success();
+    }
+
+    private void urlDecodeMap(Map<String, Object> map) throws InterruptedException, UnsupportedEncodingException {
+
+        for(Map.Entry<String, Object> entry: map.entrySet()){
+            if(Thread.currentThread().isInterrupted()) throw new InterruptedException();
+            if(entry.getValue() instanceof Map) urlDecodeMap((Map<String, Object>) entry.getValue());
+            if(isListOfMaps(entry.getValue())) urlDecodeListOfMaps((List<Map<String, Object>>) entry.getValue());
```

Comment:
using streams is not possible in this case because of InterruptedException , streams cant be used if you give it a function as lambda that is throwing checked Exception....